docs/contributing/mentorship.md

```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
+
+This process is hard and requires time, as it involves holistically looking at mentorship applications, and trying to ascertain if they would be able to benefit from this mentorship and ideally become a great future contributor or even maintainer.
+
+With several past mentorships, we have been able to distill some learnings from the process, which are listed here as recommendations to mentors,
+
+* Encourage mentees who are interested in applying, by asking them to look through our issues and trying to work on one in which they would learn a bit about the codebase!
+* Allocate some time upfront for this process, as you might have a lot of mentee applications to look through.
+  * If possible, book some time on your calendars and discuss the selection process with co-mentors in a particular quarter beforehand.
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
+    * We want to understand that the potential mentee is interested in the project and has invested some time and effort in trying to understand Thanos.
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
```

Comment:
This sounds like condition sine qua non, do we want to have it explicit here?

Reply:
So we didn't interview before, and this condition wasn't being checked, which led to some troublesome experiences. It is sine qua non, but mentioning it explicitly here, so that mentors can be mindful of this as a prerequisite. WDYT? 🙂

Reply:
All good, if there's a rationale for it as you said, then it should be here :+1: